cpp/src/arrow/json/options.h

```diff
@@ -0,0 +1,59 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef ARROW_JSON_OPTIONS_H
+#define ARROW_JSON_OPTIONS_H
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "arrow/type.h"
+#include "arrow/util/visibility.h"
+
+namespace arrow {
+
+class DataType;
+
+namespace json {
+
+struct ARROW_EXPORT ParseOptions {
+  // Parsing options
+
+  // Optional explicit schema (no type inference, ignores other fields)
+  std::shared_ptr<Schema> explicit_schema;
```

Comment:
Are we suggesting this over schema? I think it make sense to accept an optional explicit schema and optional convert/serialization transformation (string -> u64, b64 -> binary...).

Reply:
I think we are saying both. Supplying a schema is one way to initialize the conversion options

Reply:
Since this is more general than a schema, I think it'd be pretty straightforward to convert a schema into a `ConversionStrategy`. Passing a schema for `opts.fields` would be equivalent to `opts.fields =  { nm: ConvertOptions.required(typ) for (nm, typ) in zip(schm.names, schm.types) }`